tripal_chado/src/Plugin/Field/FieldType/ChadoAdditionalTypeTypeDefault.php

```diff
@@ -515,6 +515,15 @@ public static function discover(TripalEntityType $bundle, string $field_id, arra
       return $field_list;
     }
 
+    // If any instance of this field already exists on this content type,
+    // we will not suggest it for discovery. If there is a need for multiple
+    // instances, then they will need to be added manually through the GUI.
+    // Note that this will also block the existing field from showing up in
+    // the "Existing Fields" list, though.
+    if (array_key_exists(self::$id, $field_definitions)) {
+      return $field_list;
+    }
+
```

Comment:
This is what we have for `$field_definitions[self::$id]`
![PR2013 1](https://github.com/user-attachments/assets/4702be5f-0df4-406f-8703-b4b1e334baa6)

Reply:
Darn, it looks like we should have passed $entity_field_defs into the discover method for field types instead (or in addition to $all_field_defs). That would give us access to the the full field definitions for each field rather then this array form that you showed above :-(

If you are willing I would add $entity_field_defs as an additional parameter to discover here: https://github.com/tripal/tripal/blob/4.x/tripal/src/Services/TripalFieldCollection.php#L233 and then we will have access to the information we need.

This would also change the interface and the two other fields that implement the discover. For the interface, the code at https://github.com/tripal/tripal/blob/4.x/tripal/src/TripalField/Interfaces/TripalFieldItemInterface.php#L127-L152 would change to something like:

```php
  /**
   * Finds new instances of this field for a given content type.
   *
   * Fields are added to Tripal using the tripal.tripalfield_collection.*
   * configuration file during installation of a module.  In some cases,
   * however, not all possible instances of a field can be added to a content
   * type at installation of the module.  This function can be called to discover
   * if new instances of a field are appropriate for a given content type.  An
   * example of this is the `chado_property_type_default` field.  This function
   * should examine its storage backend and return a list of new fields
   * instances that could be added to the content type (i.e., bundle).
   *
   * @param \Drupal\tripal\Entity\TripalEntityType $bundle
   *   The entity type object for which new field instances should be found.
   *
   * @param string $field_id
   *   The id of the field.
   *
   * @param array $field_types
   *   An array where each item defines a field type that is attached to this 
   *   bundle. Each item is itself an array of the annotation defined at the top
   *   of that particular field type class.
   * 
   * @param array $field_instances
   *   An array of FieldConfig objects where each object defines a field attached
   *   to this content type. This differs from $field_types in that (1) there
   *   will be multiple entries if there a multiple instances of a single field
   *   type and (2) it includes the specific settings needed to create the field.
   *   As such, the FieldConfig object can be used to add existing fields to the
   *   field list returned by this method.
   *
   * @return array
   *   An associative array that follows the same structure as expected by 
   *   tripal.tripalfield_collection.* configuration.
   */
  public static function discover(TripalEntityType $bundle, string $field_id, array $field_types, array $field_instances) : array;
```

Reply:
These changes have been made in 756108ad76493a8339082c6bbc98c68cc6e952dc